File: lib-segment/src/main/java/websiteschema/mpsegment/util/CharCheckUtil.java
// Source File Name:   CharCheckUtils.java
package websiteschema.mpsegment.util;

public class CharCheckUtil {

    public static boolean isChinese(char c) {
        Character.UnicodeBlock unicodeblock = Character.UnicodeBlock.of(c);
        return unicodeblock == Character.UnicodeBlock.CJK_UNIFIED_IDEOGRAPHS;
    }

    public static boolean isChinese(String s) {
        boolean flag = true;
        char ac[] = s.toCharArray();
        for (int i = 0; i < ac.length; i++) {
            char c = ac[i];
            if (isChinese(c)) {
                continue;
            }
            flag = false;
            break;
        }

        return flag;
    }

    public static boolean isSymbol(char c) {
        int i = Character.getType(c);
        return i == 24 || i == 25 || i == 21 || i == 22 || i == 27 || i == 23 || i == 26 || i == 20;
    }
}

File: lib-segment-scala/src/main/scala/websiteschema/mpsegment/util/CharCheckUtil.scala
package websiteschema.mpsegment.util

object CharCheckUtil {

  def isChinese(c: Char): Boolean =
    Character.UnicodeBlock.of(c) == Character.UnicodeBlock.CJK_UNIFIED_IDEOGRAPHS

  def isChinese(s: String): Boolean = {
    s.find(ch => !isChinese(ch)) match {
      case None => true
      case _ => false
    }
  }

  def isSymbol(c: Char): Boolean = {
    val typo = Character.getType(c);
    return typo == 24 || typo == 25 || typo == 21 || typo == 22 || typo == 27 || typo == 23 || typo == 26 || typo == 20;
  }

  def isWhiteSpace(word: String): Boolean = word.matches("\\s+")

  def isEnglish(word: String): Boolean = word.matches("[A-z]+('|'s)?")
}
